src/components/Dialogs/DialogsTemplate.vue

```diff
@@ -0,0 +1,186 @@
+                    </the-button>
+
+                    <div v-show="formType !== 'Delete' && modifiedItem === 'Task'">
+                        <p class="mb-2 text-sm text-medium-grey dark:text-white">
+                            <span v-show="formType === 'SeeTask'">Current</span>
+                            Status
+                        </p>
+                        <custom-select
+                            :selectedStatus="selectedStatus"
+                            :columns="boardProperties.columns"
+                        />
+                    </div>
+
+                    <the-button
+                        v-show="formType !== 'SeeTask'"
+                        :regularButton="true"
+                        :isInForm="true"
+                        class="hover:bg-main-purple-hover transition-all duration-300"
+                        :class="{
+                            'button-two-bg': formType === 'Delete',
+                            'text-white bg-main-purple': formType !== 'Delete'
+                        }"
+                    >
+                        {{ buttonTwoContent }}
+                    </the-button>
+                </div>
+            </form>
+        </div>
+    </transition>
+</template>
+
+<script setup lang="ts">
+import TheButton from '../../components/shared/TheButton.vue'
+import CustomSelect from '../../components/Dialogs/elements/CustomSelect.vue'
+import SeeTask from '../../components/Dialogs/SeeTask.vue'
+import AddEdit from '../../components/Dialogs/AddEdit.vue'
+import Delete from '../../components/Dialogs/Delete.vue'
+import { returnNumberOfCompletedSubtasks } from '../../composables/completedTasks'
+import { returnBoardProperties } from '../../composables/boardProperties'
+import moreOptionsPopup from '../../composables/moreOptionsPopup'
+import { ref, Ref, computed } from 'vue'
+
+
+const props = defineProps<{
+    isDark: boolean,
+    formType: 'SeeTask' | 'Add' | 'Edit' | 'Delete',
+    modifiedItem: 'Task' | 'Board',
+    selectedStatus?: string,
+    condition: boolean
+    toggleDialog: () => void,
+    editTask: () => void,
+    editBoard: () => void,
+    deleteTask: () => void,
+    deleteBoard: () => void
+}>()
+
+const boardProperties = returnBoardProperties()
+
+const areTaskOptionsShown = ref(false)
+const { toggleOptions, closeOptions } = moreOptionsPopup
+const callMoreOptionsFn = (e: Event, cb: (e: Event, conditionToChange: Ref<boolean>) => void) => {
+  cb(e, areTaskOptionsShown)
+}
+
+const buttonOneContent = computed(() => {
+    if (props.formType === 'Delete') return 'Delete'
+
+    return `+ Add New ${props.modifiedItem === 'Task' ? 'Subtask' : 'Column'}`
+})
+
+const buttonTwoContent = computed(() => {
+    if (props.formType === 'Delete') return 'Cancel'
+    if (props.formType === 'Edit') return 'Save Changes'
+
+    const theNewWord = props.modifiedItem === 'Board' ? 'New' : ''
+    return `Create ${theNewWord} ${props.modifiedItem}`
+})
```

Comment:
could be an array of objects

Reply:
Since I intend to change this component for those blocks you mentioned yesterday, I'll leave it for now.

Reply:
Ok.